File: week_8_and_9/2_JavaScript/translate_to_js/my_solution.js
// U3.W8-9: Challenge you're converting


// I worked on this challenge [by myself, with: ].

// 2. Pseudocode
/*
create a function called separate comma with argument x
create a variable called number
set variable equal to:
	the number converted to a string
	the string split into an array

if number length is less than 4
	return the number joined into one string

if number length is less than 7
	insert a comma in the correct position

if number length is less than 10
	insert two commas in the correct positions
*/


// 3. Initial Solution
function separateComma(x) {
	var number = x.toString().split("");
	if (number.length < 4) {
		return number.join("");
	}
	else if (number.length < 7 ) {
		number.splice(number.length-3, 0, ",");
		return number.join("");
		
	}
	else if (number.length < 10) {
		number.splice(number.length-6, 0, ",");
		number.splice(number.length-3, 0, ",");
		return number.join("");
	}
}



// 4. Refactored Solution

// I could not find a way to refactor this solution.


// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE
console.log(separateComma(123) == "123")
console.log(separateComma(1234) == "1,234")
console.log(separateComma(1234567) == "1,234,567")




// 5. Reflection 

/*
I loved working on the separate_by_commas challenge, and it 
uses a lot of functions that I would like to understand in 
javascript, so I thought I would learn to do it in js!
a lot of the methods and functions are very similar, like 
join and split. Others like to_s and insert were different.
The challenge was just learning the different syntax and 
looking up the new functions, and was quite enjoyable!
*/

File: week_8_and_9/3_Ruby/fibonacci_sequence/my_solution.rb
# U3.W8-9: 


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode
# 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55
# 0, 1, 2, 3, 4, 5, 6, 7
=begin
def fib(x)
  if x == 0
    return 0
  elsif x == 1
    return 1
  elsif x == 2
    return 1
  elsif x > 2
    return fib(x-1) + fib(x-2)
  elsif x != integer
    return "not an integer"
  else
    return "not a positive number"
  end 
end


def fib(n) 
    x, y = 0, 1
    (n).times do
        x, y = y, x + y
    end  
    x
end 
=end


# 3. Initial Solution
def is_fibonacci?(num)
	case_one = Math.sqrt(5 * num**2 + 4)
	case_two = Math.sqrt(5 * num**2 - 4)
	if case_one.is_square == case_one || case_two.round == case_two
		true
	else
		false
	end
end
 puts is_fibonacci?(927372692193078999171)

# 4. Refactored Solution






# 1. DRIVER TESTS GO BELOW THIS LINE
def random_fibonacci
   [0,1,1,2,3,5,8,13,21,34,55,89,144,233,377,610,987,1597,2584,4181,6765,10946].sample
end

puts is_fibonacci?(random_fibonacci) == true
puts is_fibonacci?(8670007398507948658051921) == true
puts is_fibonacci?(random_fibonacci+100) == false
puts is_fibonacci?(927372692193078999171) == false





# 5. Reflection 